### PR TITLE
Support for key type in step-pkcs11-init

### PR DESCRIPTION
**Pull request by gdbelvin**
#### Why is this important to the project (if not answered above):
This PR enables support for RSA keys which are required for scep

#### Is there documentation on how to use this feature? If so, where?
The flags

#### In what environments or workflows is this feature supported?
All environments supported by the key API

#### Supporting links/other PRs/issues:
#583 

💔Thank you!

### CODE REVIEW

**Review by maraino — CHANGES_REQUESTED** [review]
Can you add at least --size for the number of bits in the RSA key, and optionally `--crv` to pick the EC curve to use?

**Comment by gdbelvin** [comment]
sure thing

**Comment by gdbelvin** [comment]
crv doesn't look like it's part of the API?
https://github.com/smallstep/certificates/blob/8d7c3d2f085bc4474ac022f102cd3de423aa6004/kms/softkms/softkms.go#L113

File: cmd/step-pkcs11-init/main.go [hunk 1]
```diff
@@ -297,6 +315,12 @@ func checkObject(k kms.KeyManager, rawuri string) {
 	}
 }
 
+var ktyMapping = map[string]apiv1.SignatureAlgorithm{
+	"EC":  apiv1.ECDSAWithSHA256,
+	"RSA": apiv1.SHA256WithRSAPSS,
```
Comment (maraino) [inline cmd/step-pkcs11-init/main.go:320]: Some PKCS#11 modules like YubiHSM2 can sign using `RSA-PSS`, but the same key can sign using `RSA-PKCS#1 v1.5` and it will do that by default. For example, running this code as it is will sign both certificates using PKCS#1 instead of PSS. If you want to use PSS you need to add to the templates `SignatureAlgorithm: x509.SHA256WithRSAPSS`. 

Then as YubiHSM2 can sign with both, if you want to force RSA-PSS you will need to set templates on each provisioner and manually specify the signature algorithm too.

The behavior is different for other KMSs, for example, in Google CloudKMS, if you create a key for SHA256+RSA-PSS signing, it can only sign with it, `step-ca` is able to detect this and sets the appropriate SignatureAlgorithm.

It might make sense for other KMSs to inherit the signature algorithm from the certificate if an RSA key is used, but right now it uses Go's default, in this case, RSA PKCS#1.

In summary, right now you have a couple of options, change this to be `apiv1.SHA256WithRSA` or add `SignatureAlgorithm: x509.SHA256WithRSAPSS` to the templates in this file and be aware of the known issues.

**Review by maraino — CHANGES_REQUESTED** [review]
See my comment about RSA-PSS.

File: cmd/step-pkcs11-init/main.go (hunk 1 above)
Comment (maraino) [reply cmd/step-pkcs11-init/main.go:320]: Related https://github.com/smallstep/certificates/issues/952

**Comment by maraino** [comment]
Hi @gdbelvin, have you been able to use [step-kms-plugin](https://github.com/smallstep/step-kms-plugin), can I close this as now these scripts are deprecated?

**Comment by maraino** [comment]
`step-pkcs11-init` will be removed after the v0.23.0 release.
`step` + `step-kms-plugin` can be used to create customized certificates using PKCS#11
